vault/logical_system.go

```diff
@@ -251,7 +253,27 @@ func (b *SystemBackend) handlePluginCatalogList(ctx context.Context, req *logica
 	if err != nil {
 		return nil, err
 	}
+	sort.Strings(plugins)
+	return logical.ListResponse(plugins), nil
+}
+
+func (b *SystemBackend) handlePluginCatalogUntypedList(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
```

Comment:
This might be very confusing since there's no context as to what plugin type each of the plugins might be. If somebody does a list on this endpoint and then tries to enable a aws secrets plugin when it's really an aws auth method, he/she might be confused on the error.

Instead of returning a depuped list, why not return a data response that contains three objects: `secret`, `auth`, `database` (this is similar to how listing mounts works, but will probably need to be changed to a `GET`/ `vault read`). The JSON could look something like this:

```
{
    "data": {
        "secret": [
            ...
        ],
         "auth": [
            ...
        ],
         "database": [
            ...
        ]
    }
}
```

Reply:
I think that does seem nice. I'm going to change it to that structure!